extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -600,6 +601,30 @@ public Consumer<RoutingContext> apply(String tenantId) {
                 return new Consumer<RoutingContext>() {
                     @Override
                     public void accept(RoutingContext routingContext) {
+                        OidcTenantConfig tenantConfig = routingContext.get(OidcTenantConfig.class.getName());
```

Comment:
Hi @michalvavrik This is likely not going to cover all cases, AFAIK, it is set now when either session or state cookie is used for the web-app cases, it is also set in the DefaultTenantConfigResolver - but using different keys for static/dynamic cases.

Reply:
I trust your expertise, but so far I couldn't confirm such a conclusion. It is set here:

https://github.com/quarkusio/quarkus/blob/b77999a9ba02795fc7fcf890205438043a58e1c2/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcAuthenticationMechanism.java#L88

I don't understand how is it possible to have authenticated OIDC tenant without having OIDC tenant config resolved.

I'll be grateful for any hint how should this be implemented. Thank you.

Reply:
@michalvavrik Oh yes, I forgot I added it recently when handling session cookie chunk issues, I guess what I wanted to say that there is some duplication there (which I'll need to fix), but this is indeed the most precise check